### PR TITLE
feat: hashed link retrieval from event types v1 api

### LINKED ISSUE #15619: [CAL-4047] Retrieve private URL or hashedLink through event-types API(s)
### Is your proposal related to a problem?

Yes, this is a blocking issue to copy private urls from Raycast:
* related https://github.com/raycast/extensions/issues/13215
* related https://github.com/calcom/cal.com/issues/5106

### Describe the solution you'd like

We'd like a way to retrieve the hashedLink for the private URL through the APIs, either as:
* new key-value pair in the `GET /event-types` or `GET /event-types/{id}` response
* replacing the current `link` value in the response with the private URL in case user has it enabled.
* ...other alternative like a new API.

### Describe alternatives you've considered

No alternatives that I could find in the [API reference](https://cal.com/docs/enterprise-features/api/api-reference#the-basics).

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

<sub>[CAL-4047](https://linear.app/calcom/issue/CAL-4047/retrieve-private-url-or-hashedlink-through-event-types-apis)</sub>

### PR DESCRIPTION
**Pull request by JOSU9435**
## What does this PR do?

- Fixes #15619 (GitHub issue number)
- Fixes CAL-4047 (Linear issue number)

Adds an optional `hashedLink` field in the event-types v1 APIs response for retrieving the private URL.

```json
"hashedLink": {
    "link": "cS4DSPAq28ALSHkyFov1y2"
}
```

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs).
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Start the v1 API and generate an API key.
- Send a GET request to http://localhost:3003/api/v1/event-types?apiKey={API_KEY}. The response should include the hashedLink field.
- Send a GET request to http://localhost:3003/api/v1/event-types/{eventTypeId}?apiKey={API_KEY}. The response should contain the hashLinked field.

### CODE REVIEW

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Thanks for the PR @JOSU9435 but I have a suggestion
Let's not return eventTypeId as it's already present. Also remove id of hashed link as there is nothing that can be done with it.

File: apps/api/v1/pages/api/event-types/_get.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ async function getHandler(req: NextApiRequest) {
     },
     include: {
       customInputs: true,
+      hashedLink: true,
```
Comment (anikdhabal) [inline apps/api/v1/pages/api/event-types/_get.ts:59]: Same

File: apps/api/v1/pages/api/event-types/[id]/_get.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export async function getHandler(req: NextApiRequest) {
     where: { id },
     include: {
       customInputs: true,
+      hashedLink: true,
```
Comment (anikdhabal) [inline apps/api/v1/pages/api/event-types/[id]/_get.ts:54]: Select only the link here

**Review by anikdhabal — DISMISSED** [review]
LGTM !!

**Comment by JOSU9435** [comment]
Thanks for the review @Amit91848, @anikdhabal 
I have pushed the changes that you have suggested.

**Review by anikdhabal — APPROVED** [review]

**Review by Amit91848 — APPROVED** [review]
LGTM!!

**Pull request merged**